manticore/manticore.py

```diff
@@ -628,6 +628,7 @@ def terminate(self):
         a :func:`~hook`.
         '''
         self._executor.shutdown()
+        raise TerminateState('Terminated')
```

Comment:
This will cancel exploration of current state. 
And it will wait until next fork for all the other running states already in a worker.